core/api/extrinsic/extrinsic_jrpc_processor.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_SERVICE_EXTRINSICS_SUBMISSION_SERVICE_HPP
+#define KAGOME_CORE_SERVICE_EXTRINSICS_SUBMISSION_SERVICE_HPP
+
+#include <memory>
+#include <vector>
+
+#include <jsonrpc-lean/server.h>
+#include <boost/core/noncopyable.hpp>
+#include "api/extrinsic/extrinsic_api.hpp"
+#include "api/jrpc/jrpc_processor.hpp"
+
+namespace kagome::api {
+
+  /**
+   * @brief extrinsic submission service implementation
+   */
+  class ExtrinsicJRPCProcessor
+      : public JRPCProcessor,
+        public std::enable_shared_from_this<ExtrinsicJRPCProcessor>,
+        private boost::noncopyable {
+   public:
+    /**
+     * @brief constructor
+     */
+    explicit ExtrinsicJRPCProcessor(std::shared_ptr<ExtrinsicApi> api);
+
+   private:
+    std::shared_ptr<ExtrinsicApi> api_;  ///< api implementation
+  };
```

Comment:
what is the reason of creating this class? underlying class `JRPCProcessor` has non-virtual methods

Reply:
There are another apis, which probably will need to be implemented. JRPCProcessor contains common methods and should be the base class for such api processors.